### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by mwhamgenomics**
- PEP8 refactor
- refactoring datatable configs and HTML templates
- adding page rendering tests for web app endpoints
- adding tests for charts.js

### CODE REVIEW

File: reporting_app/__init__.py [hunk 1]
```diff
@@ -15,17 +15,14 @@
 version = open(join(dirname(dirname(__file__)), 'version.txt')).read()
 
 
-def construct_url(endpoint, **query_args):
-    url = rest_api().api_url(endpoint)
-    if query_args:
-        url += '?' + '&'.join(['%s=%s' % (k, v) for k, v in query_args.items()])
-    return url.replace('\'', '"')
-
-
```
Comment (mwhamgenomics) [inline reporting_app/__init__.py]: Moved to `util`

File: reporting_app/templates/login.html [hunk 2]
```diff
@@ -16,7 +16,6 @@
         </div>
     </div>
 </div>
-{{ super() }}
```
Comment (mwhamgenomics) [inline reporting_app/templates/login.html]: Login view now overrides reporting_base's navbar

File: reporting_app/templates/proc_report.html [hunk 3]
```diff
@@ -1,26 +1,6 @@
-{% macro proc_header() -%}
-<script src="//cdnjs.cloudflare.com/ajax/libs/vis/4.19.1/vis.min.js"></script>
-<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/vis/4.19.1/vis.min.css" type="text/css" />
-<script src="//cdnjs.cloudflare.com/ajax/libs/moment.js/2.18.1/moment.min.js"></script>
-<script src="/static/stage_timeline.js"></script>
```
Comment (mwhamgenomics) [inline reporting_app/templates/proc_report.html]: Moved to `head` block in reporting_base

File: reporting_app/templates/proc_report.html [hunk 4]
```diff
@@ -56,24 +36,9 @@
...
-                <li class="nav"><a data-target="#analysis_driver_proc_{{ loop.index + 1 }}_tab" data-toggle="tab">{{ proc._created }}</a></li>
-            {% endfor %}
-        </ul>
-
-        <div id="analysis_driver_proc_1_tab" class="tab-pane fade in active">{{ proc_timeline(procs[0]) }}</div>
-        {% for proc in procs[1:] %}
-            <div id="analysis_driver_proc_{{ loop.index + 1 }}_tab" class="tab-pane fade">{{ proc_timeline(proc) }}</div>
-        {% endfor %}
-    {% endif %}
-</div>
```
Comment (mwhamgenomics) [inline reporting_app/templates/proc_report.html]: This is now a generic macro

File: reporting_app/templates/reporting_base.html [hunk 5]
```diff
@@ -1,74 +1,54 @@
...
-                        <li><a href="/project_status/all">All Projects Status</a></li>
-                    </ul>
-                </li>
-                <li><a href="/charts">Charts</a></li>
-                <li class="dropdown">
-                    <a href="#" class="dropdown=toggle" data-toggle="dropdown">{{ current_user.username }}<b class="caret"></b></a>
-                    <ul class="dropdown-menu">
-                        <li><a href="/change_password">Change password</a></li>
-                        <li><a href="/logout">Logout</a></li>
-                    </ul>
```
Comment (mwhamgenomics) [inline reporting_app/templates/reporting_base.html]: Moved to base.html

File: reporting_app/templates/tabbed_datatables.html [hunk 6]
```diff
@@ -1,29 +0,0 @@
...
-            {% endfor %}
-        </ul>
-
-        <div id="{{ tab_set.tables[0].name }}_tab" class="tab-pane fade in active">{{ datatable(tab_set.tables[0], False) }}</div>
-        {% for table in tab_set.tables[1:] %}
-            <div id="{{ table.name }}_tab" class="tab-pane fade">{{ datatable(table, False) }}</div>
-        {% endfor %}
-    {% endif %}
-</div>
-{% endfor %}
```
Comment (mwhamgenomics) [inline reporting_app/templates/tabbed_datatables.html]: Moved to tab_sets.html

File: reporting_app/util.py [hunk 7]
```diff
@@ -1,13 +1,32 @@
...
     direction = 'desc' if col_name.startswith('-') else 'asc'
     return [[c['data'] for c in cols].index(col_name.lstrip('-')), direction]
 
 
-def datatable_cfg(title, cols, api_url=None, ajax_call=None, default_sort_col=None, **kwargs):
+def construct_url(endpoint, **query_args):
+    url = flask_login.current_user.comm.api_url(endpoint)
+    if query_args:
+        _query_args = []
+        for k in sorted(query_args):
```
Comment (mwhamgenomics) [inline reporting_app/util.py]: Mkaing construct_url deterministic

File: reporting_app/templates/proc_report.html [hunk 8]
```diff
@@ -56,24 +36,9 @@
...
-
+<br/>
+<p>Number of times processed: {{ procs|length }}</p>
+{% call(tab) tabset('analysis_driver_procs', 'Analysis Driver processes', procs, '_created') %}
```
Comment (tcezard) [inline reporting_app/templates/proc_report.html]: The new marco is showing the analysis_driver_procs section by default where it used to be hidden.
Adding an option to that effect in the macro should fix this
```python
tabset('analysis_driver_procs', 'Analysis Driver processes', procs, '_created', default_collapsed='')
...
{% macro tabset(name, title, tabs, title_field='title', default_collapsed='in') -%}
...
<div id="{{ name }}_tabs" class="tab-content collapse {{ default_collapsed }}">

```

**Pull request merged**